src/octotree.js

```diff
@@ -24,9 +25,23 @@ $(document).ready(() => {
       if (event.target === window) layoutChanged();
     });
 
-    $toggler.click(toggleSidebarAndSave);
+    $pinner.click(togglePinAndSave);
+
+    $sidebar.on('keyup mouseleave mousemove', debounce({
+      callback: () => toggleSidebar(false),
+      cancelIf: ({type}) => isSidebarPinned() || type === 'mousemove',
```

Comment:
Avoid destructuring here. Change {type} to event, so it's clearer what this method takes in. (Not that I'm against destructuring, but this is a special case as debounce() return a fn(...args); so there must be a place that is explicit about the param type.)